Packs/Gmail/Integrations/Gmail/Gmail.py

```diff
@@ -745,12 +758,109 @@ def get_millis_from_date(date, arg_name):
             raise ValueError(f'{arg_name} argument is not in expected format.')
 
 
+def cutting_for_batches(list_accounts: list) -> List[list]:
+
+    accounts: list = []
+    rest_accounts: list = []
+
+    batch_size = int(len(list_accounts) / BATCH_DIVIDER)
+    if rest := len(list_accounts) % BATCH_DIVIDER:
+        rest_accounts = list_accounts[-rest:]
+        list_accounts = list_accounts[:-rest]
+
+    accounts.extend(batch(list_accounts, batch_size))
+
+    if rest_accounts:
+        accounts[0].extend(rest_accounts)
+
+    return accounts
+
+
+def scheduled_commands_for_more_users(accounts: list, next_page_token: str) -> List[CommandResults]:
+
+    accounts_batches = cutting_for_batches(accounts)
+
+    command_results: list[CommandResults] = []
+    args = copy.deepcopy(demisto.args())
+    for batch in accounts_batches:
+
+        args.update({'list_accounts': batch})
+        command_results.append(
+            CommandResults(
+                readable_output='Serching mailboxes, please wait...',
+                scheduled_command=ScheduledCommand(
+                    command='gmail-search-all-mailboxes',
+                    next_run_in_seconds=10,
+                    args=copy.deepcopy(args),
+                    timeout_in_seconds=600
+                )
+            )
+        )
+        args.pop('list_accounts', None)
+    if next_page_token:
+        command_results.append(
+            CommandResults(
+                outputs_key_field='PageToken',
+                outputs={'PageToken': {'NextPageToken': next_page_token}}
+            )
+        )
+
+    return command_results
+
+
+def get_mailboxes(max_results: int, users_next_page_token: str = None):
+    '''
+    Used to fetch the list of accounts for the search-all-mailboxes command
+    '''
+    list_accounts: list[str] = []
+    counter = 0
```

Comment:
what are we counting? Let's name it `<...>_count` according to the thing being counted